dpnp/dpnp_iface_statistics.py

```diff
@@ -291,8 +289,14 @@ def cov(x1, y=None, rowvar=True, bias=False, ddof=None, fweights=None, aweights=
         elif aweights is not None:
             pass
         else:
-            if x1_desc.dtype != dpnp.float64:
-                x1_desc = dpnp.get_dpnp_descriptor(dpnp.astype(x1, dpnp.float64), copy_when_nondefault_queue=False)
+            if not rowvar and x1.shape[0] != 1:
+                x1 = x1.get_array() if isinstance(x1, dpnp_array) else x1
+                x1 = dpnp_array._create_from_usm_ndarray(x1.mT)
+                x1 = dpnp.astype(x1, dpnp.float64) if x1_desc.dtype != dpnp.float64 else x1
```

Comment:
Please test that this works on Iris Xe, or other devices that do not support `float64` data type.

Reply:
It will not work (and in fact it was not working previously as well). @antonwolfy mentioned it is written like this because Numpy by default returns `float64` for any input data type. The suggestion was to fix it in a separate ticket since it needs to correct dpnp c++ backend code.

Reply:
I've filed #1379 to add support of `dpnp.cov` on Iris Xe